src/service/xhr-impl.js

```diff
@@ -222,7 +226,7 @@ export class Xhr {
           init.body
       );
       // Content should be 'text/plain' to avoid CORS preflight.
-      init.headers['Content-Type'] = init.headers['Content-Type'] ||
+      init.headers['Content-Type'] = headerContentType ||
```

Comment:
So, this is JSON case and it's super important that "Content-Type" is "text/plain" in this case. No other content type is allowed for JSON to avoid triggering CORS preflight. So please revert this line.